src/tcgwars/logic/impl/gen4/RisingRivals.groovy

```diff
@@ -521,16 +678,30 @@ public enum RisingRivals implements LogicCardInfo {
         return evolution (this, from:"Kakuna", hp:HP110, type:GRASS, retreatCost:1) {
           weakness R, PLUS30
           pokePower "Flutter Wings", {
-            text "Once during your turn , you may search your deck for a Pokémon, show it to your opponent, and put it into your hand. Shuffle your deck afterward. This power can’t be used if Beedrill is affected by a Special Condition."
+            text "Once during your turn , you may search your deck for a [G] Pokémon, show it to your opponent, and put it into your hand. Shuffle your deck afterward. This power can’t be used if Beedrill is affected by a Special Condition."
             actionA {
+              checkLastTurn()
+              checkNoSPC()
+              assert my.deck : "Your deck is empty"
+              powerUsed()
+              my.deck.search("Search your deck for a [G] Pokémon",pokemonTypeFilter(G)).showToOpponent("Flutter Wings: Selected card").moveTo(my.hand)
+              shuffleDeck()
             }
           }
           move "Needle Shock", {
             text "30 damage. The Defending Pokémon is now Paralyzed and Poisoned. Ignore this effect if any of your Pokémon used Needle Shock during your last turn."
             energyCost G, C, C
-            attackRequirement {}
             onAttack {
-              damage 0
+              damage 30
+              afterDamage {
+                bc"${bg.em().retrieveObject("Needle_Shock")}"
+                bc"$bg.turnCount"
+                if(bg.em().retrieveObject("Needle_Shock")!=(bg.turnCount - 2)) {
```

Comment:
The key needs to include the player, otherwise your opponent using Needle Shock will overwrite you using it last turn.